cot/benches/bench_utils.rs

```diff
@@ -0,0 +1,243 @@
+//! Cot application benchmarking utilities using Criterion.
+//!
+//! This module provides scaffolding for benchmarking Cot-based web servers,
+//! supporting both independent benchmarks and grouped benchmarks.
+//!
+//! **Note:** Requires the `test` feature to be enabled.
+//! Run with: `cargo bench --features test`
+
+use cot::cli::CliMetadata;
+use cot::config::ProjectConfig;
+use cot::project::RegisterAppsContext;
+use cot::router::Router;
+use cot::{App, AppBuilder, Project};
+use criterion::{Criterion, Throughput};
+use futures_util::future::join_all;
+
+pub(crate) struct ProjectBenchmarkBuilder<'a> {
+    #[allow(clippy::allow_attributes, dead_code)]
+    name: &'static str,
+    #[allow(clippy::allow_attributes, dead_code)]
+    criterion: &'a mut Criterion,
+    method: reqwest::Method,
+    path: Option<&'static str>,
+    body: Option<String>,
+    content_type: Option<&'static str>,
+    expected_status_code: reqwest::StatusCode,
+    requests_per_iteration: u64,
+}
+
+macro_rules! builder_method {
+    ($name:ident, $ty:ty) => {
+        #[must_use]
+        pub(crate) fn $name(mut self, $name: $ty) -> Self {
+            self.$name = $name;
+            self
+        }
+    };
+    ($name:ident, $ty:ty, opt) => {
+        #[must_use]
+        pub(crate) fn $name(mut self, $name: $ty) -> Self {
+            self.$name = Some($name);
+            self
+        }
+    };
+}
+
+#[allow(clippy::allow_attributes, dead_code)]
```

Comment:
I'm wondering if we could remove the `#[allow(dead_code)]` attributes, possibly by introducing a feature flag.

If not, can we add `reason = "..."` for each attribute?